draft-ietf-tls-ctls.md

```diff
@@ -139,7 +136,52 @@ where the most complexity and size reduction can be obtained. Most of the other
 TLS 1.3 are highly optimized and do not require compression to be used.
 
 The compression profile defining the use of algorithms, algorithm parameters, and
-extensions is specified via a JSON dictionary.
+extensions is represented by the `CTLSTemplate` structure:
+
+~~~~
+enum {
+  profile(0),
+  version(1),
+  cipher_suite(2),
+  dh_group(3),
+  signature_algorithm(4),
+  random(5),
+  mutual_auth(6),
+  handshake_framing(7),
+  client_hello_extensions(8),
+  server_hello_extensions(9),
+  encrypted_extensions(10),
+  cert_request_extensions(11),
+  known_certificates(12),
+  finished_size(13),
+  optional(65535)
+} CTLSTemplateElementType;
+
+struct {
+  CTLSTemplateElementType type;
+  opaque data<0..2^32-1>;
+} CTLSTemplateElement;
+
+struct {
+  CTLSTemplateElement elements<0..2^32-1>
+} CTLSTemplate;
+~~~~
+
+> TODO: Reorder enum.
+
+Elements in a `CTLSTemplate` MUST appear in strictly ascending order.
```

Comment:
```suggestion
Elements in a `CTLSTemplate` MUST appear sorted by the type field in ascending order.
```

Reply:
Done (but kept "strictly" due to uniqueness requirements).